src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/MetadataAggregatorTests.cs

```diff
@@ -175,6 +179,69 @@ public void HeapSizes()
             TestGenerationHandle(aggregator, MetadataTokens.GuidHandle(3), expectedHandle: MetadataTokens.GuidHandle(3), expectedGeneration: 4);
 
             AssertExtensions.Throws<ArgumentException>("handle", () => TestGenerationHandle(aggregator, MetadataTokens.StringHandle(22), expectedHandle: MetadataTokens.StringHandle(0), expectedGeneration: 0));
+            // Sizes are not cumulative for GUIDs. They represent the number of available GUIDs.
+            AssertExtensions.Throws<ArgumentException>("handle", () => TestGenerationHandle(aggregator, MetadataTokens.GuidHandle(4), expectedHandle: MetadataTokens.StringHandle(0), expectedGeneration: 0));
+        }
+
+        [Fact]
+        public void HeapSize_GuidHeapSizes()
+        {
+            MetadataReader baseReader = MetadataReaderTests.GetMetadataReader(NetModule.AppCS); ;
+            var bytes = CreateMinimalReaderMetadataAsBytes(Guid.NewGuid());
+            unsafe
+            {
+                fixed (byte* pointer = bytes)
+                {
+                    MetadataReader mr1 = CreateMinimalReaderWithGuid(pointer, bytes.Length);
+                    MetadataReader mr2 = CreateMinimalReaderWithGuid(pointer, bytes.Length);
+                    IReadOnlyList<MetadataReader> metadataReaders = new List<MetadataReader> { mr1, mr2 };
+
+                    var aggregator = new MetadataAggregator(baseReader, metadataReaders);
+
+                    var guidSizeReader0 = baseReader.GetHeapSize(HeapIndex.Guid) / 16;
+                    Assert.Equal(1, guidSizeReader0);
+                    var guidSizeReader1 = mr1.GetHeapSize(HeapIndex.Guid) / 16;
+                    Assert.Equal(1, guidSizeReader1);
+                    var guidSizeReader2 = mr2.GetHeapSize(HeapIndex.Guid) / 16;
+                    Assert.Equal(1, guidSizeReader2);
+
+                    TestGenerationHandle(aggregator, MetadataTokens.GuidHandle(1), expectedHandle: MetadataTokens.GuidHandle(1), expectedGeneration: 0);
+
+                    // GUID-heap allocation shouldn't be cumulative, since GUIDs are copied among generations.
+                    // The delta-reader above need indeed a single GUID allocation in each gen.
+                    AssertExtensions.Throws<ArgumentException>("handle", () => TestGenerationHandle(aggregator, MetadataTokens.GuidHandle(2), expectedHandle: MetadataTokens.GuidHandle(2), expectedGeneration: 0));
```

Comment:
We should be able to read 3 guids since there is one in baseline, and one in each delta, no?

Reply:
OK